search/includes/classes/class-queue.php

```diff
@@ -612,11 +641,10 @@ public function checkout_jobs( $count = 250 ) {
 		$table_name = $this->schema->get_table_name();
 
 		// TODO transaction
-		// TODO only find objects that aren't already running
 
 		$jobs = $wpdb->get_results(
 			$wpdb->prepare(
-				"SELECT * FROM {$table_name} WHERE ( `start_time` <= NOW() OR `start_time` IS NULL ) AND `status` = 'queued' LIMIT %d", // Cannot prepare table name. @codingStandardsIgnoreLine
+				"SELECT * FROM {$table_name} WHERE ( `start_time` <= NOW() OR `start_time` IS NULL ) AND `status` = 'queued' ORDER BY `job_id` LIMIT %d", // Cannot prepare table name. @codingStandardsIgnoreLine
```

Comment:
Do we need to have the tasks order by its id? I get that's the way it was, but unless we need to guarantee that property, we can save an expensive `ORDER BY`.

Reply:
Nah I added the Order by. For the lowest-highest range to work it needs to be ordered otherwise there could be conflicts between what different parallel processors will take. (one would have `1-10` from `[1,2,10]` and another one would have `5-20` from `[5,6,20]`, both of them would then try to process jobs `5,6,10`)